src/Cache/Engine/RedisEngine.php

```diff
@@ -228,14 +228,22 @@ public function clear($check)
         if ($check) {
             return true;
         }
-        $keys = $this->_Redis->getKeys($this->_config['prefix'] . '*');
 
-        $result = [];
-        foreach ($keys as $key) {
-            $result[] = $this->_Redis->del($key) > 0;
-        }
+        $result = true;
+        $iterator = null;
+        $pattern = $this->_config['prefix'] . '*';
+        do {
+            $keys = $this->_Redis->scan($iterator, $pattern);
+
+            // Redis may return empty results, so protect against that
+            if ($keys !== false) {
+                foreach($keys as $key) {
+                    $result = $result && ($this->_Redis->del($key) > 0);
```

Comment:
After `$result` becomes `false`, `del()` won't be called.

Either return `false` immediately (but this would be different behavior), or use something similar to previous code.

Reply:
Ah stupid, thanks for the feedback. I'll fix this, so this behavior doesn't change.